addons/event/models/event.py

```diff
@@ -390,11 +390,15 @@ def _onchange_partner(self):
     @api.multi
     def message_get_suggested_recipients(self):
         recipients = super(EventRegistration, self).message_get_suggested_recipients()
+        public_users = self.env["res.users"]
+        public_groups = self.env.ref("base.group_public", False)
+        if public_groups:
+            public_users = public_groups.mapped("users")
         try:
             for attendee in self:
-                if attendee.partner_id:
+                if attendee.partner_id and not attendee.partner_id.user_ids & public_users:
                     attendee._message_add_suggested_recipient(recipients, partner=attendee.partner_id, reason=_('Customer'))
-                elif attendee.email:
+                if attendee.email not in {False, attendee.partner_id.email}:
```

Comment:
Can't we keep the elif ? Either we have a partner that is not public, either we use the email.

Reply:
Usually the email gets synced in an onchange from the partner, if there's one. So if it's actually different, then it must be because the user intentionally changed it. He would expect it to appear there then, along with the other suggestion.

Reply:
I agree with your comment. In all message_get_suggested_recipients override in standard code we take the partner, and if not set the email, even if emails are different. The main reason behind that is that the customer is more important, as he is probably your real contact and he will probably be added in the followers list. Having different emails is more linked to an outdated record. We consider that partner > email. Another purpose is also to avoid bloating the interface with several emails and/or creating new partners if the email is not linked to an existing partner.

As this PR targets a stable version I would stick on this convention, partner or email but not both. That way the behavior is consistent through various addons that override message_get_suggested_recipients.

Does that seem ok to you ?